### ISSUE TITLE
Ligatures not working for specific font (LobsterTwo-Bold.otf)

### ISSUE THREAD

**Issue by verteks16** [opentypejs/opentype.js#590]
## Expected Behavior

Opentype.js should output text with ligatures for getPath() conversion.
## Current Behavior

Result of font.getPath() call results in text which doesn't have ligatures, even if font supports them.

I explicitly request 'liga' feature in getPath() call:
```
font.getPath(
  htmlBlock.text, 
  0, 
  0, 
  htmlBlock.fontSize, 
  {
    letterSpacing: htmlBlock.letterSpacing,
    features: {
      'liga': true,
    }
  }
);
```

Release 1.3.4 converts to path silently changing ligatures to standard letters.

I tried @Connum  fork of opentype.js, branch which fixes #554 (PR: https://github.com/opentypejs/opentype.js/pull/564) - this throw error "Substitution type 11 is not supported in chaining substitution". Seems logical, this branch adds check:
```
if (substitutionType === '12') {
  <lookup code>
} else {
  throw new Error(`Substitution type ${substitutionType} is not supported in chaining substitution`);
}
```
around substitution parsing.

FontForge reports this font contains mutiple definitions of 'liga' lookups:
![image](https://user-images.githubusercontent.com/26660595/226576202-de970e6c-6a3d-4c6d-aaff-92140cd780f1.png)

To see the difference look at provided example, more informations about it in 'Steps to Reproduce' section.
## Possible Solution

No idea, I'm newbie in font world :)
## Steps to Reproduce (for bugs)

I provide minimal example - upper text is how browser renders this text (with ligatures), lower text is upper text converted by opentype.js conversion to path commands. Notice different letters 'd', 'u' and 'fft' ligature...

... or just look at this screenshot:
![image](https://user-images.githubusercontent.com/26660595/226573953-90c4ca52-82e3-44c6-a50d-567bb09145ff.png)

## Context

I try to use opentype.js as a tool to convert HTML text to SVG path elements. I treat opentype.js as 'magic box'

## Your Environment

* Version used: 1.3.4
* Font used: LobsterTwo-Bold.otf (provided in ZIP file)
* Browser Name and version: Chromium 78 / Chrome 103
* Operating System and version (desktop or mobile): Ubuntu 22.04 LTS
* Link to your project: [minimalexample.zip](https://github.com/opentypejs/opentype.js/files/11026888/minimalexample.zip)

**Comment by st0nerhat** [comment]
The exception being thrown here was added in PR #486. I believe it's having the desired effect which is crashing on an unsupported substitution instead of silently substituting incorrectly, which is what happens if the exception is commented out. Let me take a look at what substitution type '11' is.

**Comment by st0nerhat** [comment]
Btw, totally open to changing this to a warning instead.

**Comment by st0nerhat** [comment]
Ok the relevant table type is https://learn.microsoft.com/en-us/typography/opentype/spec/gsub#11-single-substitution-format-1

**Comment by st0nerhat** [comment]
So yeah, this is just another gap in opentype.js's [chained sequence context format 3](https://learn.microsoft.com/en-us/typography/opentype/spec/chapter2#chained-sequence-context-format-3-coverage-based-glyph-contexts) implementation. I don't think it would be particularly hard to add, given that it's the absolute simplest form of substitution, but nevertheless it is missing.

**Comment by st0nerhat** [comment]
`getLookupMethod` supports `11` lookup tables amongst others, should the `if` statement on line 165 be removed? The logic inside there should work as is for any of formats. `getLookupMethod` already throws if it's passed a format that it doesn't support.

**Comment by Connum** [comment]
Yeah, that's what I thought. However, it then always uses the replacement forms, because we're not actually checking the backtrack and lookahead context. Going on the commit message, I guess chaining substitution was never implemented correctly and it was more of a hotfix for some ligatures used in Arabic scripts.

**Comment by verteks16** [comment]
@st0nerhat This exception is totally fine for me, even more correct when I explicitly tell opentype.js to use 'liga' feature. Font contains those ligatures but opentype.js can't use them properly in current state.

I tested more solutions to my task (text -> path conversion). One more comparison:
HTML <p> tag (correct output) :
![image](https://user-images.githubusercontent.com/26660595/227468150-3bf25835-209e-4e64-9301-c67de6791d4b.png)

Opentype.js (incorrect output):
![image](https://user-images.githubusercontent.com/26660595/227468046-0bc6c0de-8bdd-41d8-bc64-f1ab6b0c74e0.png)

Typr.js (incorrect output):
![image](https://user-images.githubusercontent.com/26660595/227466902-d3e97808-f1db-47cd-9af0-b07710bc72cf.png)

Typr.js + HarfBuzz.js [[explanation]](https://github.com/photopea/Typr.js#shaping-with-harfbuzz) (correct output):
![image](https://user-images.githubusercontent.com/26660595/227467148-fd104b90-7aec-46d1-bf8a-0c450c82cbc2.png)

I would call this piece of code "partially implemented" rather than "incorrect". Before `contextRulesMatch` there are some lookahead / backtrack calculations for Tashkeel Arabic characters, whatever they are.
